server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -481,4 +482,53 @@ public void onFailure(Exception e) {
             }
         };
     }
+
+    public void startRecommissionAction(final ActionListener<AcknowledgedResponse> listener) {
+        /*
+         * For abandoned requests, we might not really know if it actually restored the exclusion list.
+         * And can land up in cases where even after recommission, exclusions are set(which is unexpected).
+         * And by definition of OpenSearch - Clusters should have no voting configuration exclusions in normal operation.
+         * Once the excluded nodes have stopped, clear the voting configuration exclusions with DELETE /_cluster/voting_config_exclusions.
+         * And hence it is safe to remove the exclusion if any. User should make conscious choice before decommissioning awareness attribute.
+         */
+        decommissionController.clearVotingConfigExclusion(new ActionListener<Void>() {
```

Comment:
should we provide an option to not force clear voting as a flag to the API?

Reply:
Cluster might get into an unexpected state if we don't clear the exclusion and recommissioned the zone. When the cluster decommissioned one zone, it was expected that no further exclusion will be set as we already decommissioned one master. Any exclusion after decommissioning, would just be making things worse for a cluster already under stress. Not sure, if the user would need any exclusion set when the recommissioning the zone back.

Also as per discussion in #4084 PR, the DELETE API should try to restore the system back for multiple failure cases

Let me know if you see a case where recommissioning won't expect the exclusion to be wiped off